src/style-property-map.js

```diff
@@ -14,16 +14,18 @@
 
 (function(shared, scope, testing) {
 
-function StylePropertyMap() {}
+function StylePropertyMap(styleObject) {
+  this._styleObject = styleObject;
+}
 
-  StylePropertyMap.prototype = {
-    append: function(property, value) {},
-    delete: function(property) {},
-    get: function(property) {},
-    getAll: function(property) {},
-    has: function(property) {},
-    set: function(property, value) {},
-    getProperties: function() {}
+  StylePropertyMap.prototype.prototype = StylePropertyMapReadOnly.prototype;
```

Comment:
Right now you're adding stuff to the ReadOnly one's prototype too. I have some ideas about how to fix the prototype chain which I'll look into as well, but in the meantime it's best to make a copy.